variant/param_name_variant.js

```diff
@@ -0,0 +1,65 @@
+// This script provides a mechanism whereby the active scanner can inject payloads appended to the name of a parameter.
+// This might be handy, for example: If you wanted to test for NoSQL Injections such as username=exampleuser&password[$ne]
+// This script is designed to work with starard URL Query Parameters (GET) 
+// and application/x-www-form-urlencoded (POST) parameters.
+
+// The parseParameter function will typically be called for every page and 
+// the setParameter function is called by each active plugin to bundle specific attacks
+
+// Note that new custom input vector scripts will initially be disabled
+// Right click the script in the Scripts tree and select "enable"
+
+// Declare classes used herein
+var VariantURLQuery = Java.type("org.parosproxy.paros.core.scanner.VariantURLQuery");
+var VariantFormQuery = Java.type("org.parosproxy.paros.core.scanner.VariantFormQuery");
+var NameValuePair = Java.type("org.parosproxy.paros.core.scanner.NameValuePair");
+var vURLQuery = new VariantURLQuery();

```

Comment:
Create the instances just in `parseParameters` and later in `setParameter` use them instead of checking the method each time?